src/local/butler/scripts/github_metadata_backfiller.py

```diff
@@ -0,0 +1,102 @@
+
+from clusterfuzz._internal.datastore import data_types
+from libs.issue_management import issue_tracker_utils
+from libs.issue_management import oss_fuzz_github
+
+
+def _get_testcase(bug_information):
+  """Retrieve testcase based on its bug information."""
+  candidates = list(
+      data_types.Testcase.query(
+          data_types.Testcase.bug_information == bug_information))
+  if not candidates:
+    print('No candidate found when '
+          'querying bug information {bug_information}\n')
+  elif len(candidates) > 1:
+    print('Multiple candidates found when '
+          f'querying bug information {bug_information}:\n')
+    for testcase in candidates:
+      print(f'  {testcase.key.id()}')
+  else:
+    return candidates[0]
+  return None
+
+
+def _get_bug_information(issue):
+  """Given a GitHub issue, parse its corresponding bug information."""
+  bug_information = issue.title[len(oss_fuzz_github.ISSUE_TITTLE_TEXT_PREFIX) +
+                                1:]
+  if bug_information.isdigit():
+    return bug_information
+  return None
+
+
+def _testcase_information_verified(testcase, issue):
+  """Verify if a testcase correctly stores its GitHub issue information."""
+  if testcase.github_repo_id == issue.repository.id and \
+          testcase.github_issue_num == issue.number:
+    print(f'Testcase {testcase.bug_information} was properly stored.')
+    return True
+
+  if testcase.github_repo_id is None and testcase.github_issue_num is None:
+    print(f'Testcase {testcase.bug_information} was not stored.')
+  else:
+    print(f'Testcase {testcase.bug_information} stored '
+          f'is inconsistent with GitHub:\n'
+          f'  Issue number  (Storage) {testcase.github_issue_num} '
+          f'!= {issue.number} (GitHub)\n'
+          f'  Repository ID (Storage) {testcase.github_repo_id} '
+          f'!= {issue.repository.id} (GitHub).')
+  return False
+
+
+def execute(args):
+  """Backtrack GitHub issues filed in the past,
+    update their information in gcloud, and close them when necessary."""
+  issue_tracker = issue_tracker_utils.get_issue_tracker('oss-fuzz')
+  for issue in oss_fuzz_github.get_my_issues():
+    print('========================================')
+    # Track testcase.
+    bug_information = _get_bug_information(issue)
+    if not bug_information:
+      print('Unable to extract bug information: '
+            f'Repo {issue.repository.id} Issue {issue.number}.\n'
+            f'Issue title: {issue.title}.')
+      continue
+    testcase = _get_testcase(bug_information)
+
+    # Backclose issues.
+    monorail_issue = issue_tracker.get_original_issue(testcase.bug_information)
+    if not monorail_issue.is_open:
+      print(f'Closing testcase (bug information: {testcase.bug_information}):\n'
+            f'  Issue number  {issue.number}\n'
+            f'  Repository ID {issue.repository.id}\n')
+      if args.non_dry_run:
+        oss_fuzz_github.close_issue(testcase)
```

Comment:
Looking at the implementation of `close_issue`, it looks like we early exit if there's no issue info set:

```
  if not _issue_recorded(testcase):                                                                                           
    return  
```

It might be better after all to have your old order after all, and instead of continuing when `_testcase_information_verified`, we do the testcase.put() if necessary.

i.e.

```python
    if not _testcase_information_verified(testcase, issue):
      # Instead of continuing
      print(f'Updating testcase (bug information: {testcase.bug_information}):\n'
            f'  Issue number  {issue.number}\n'
            f'  Repository ID {issue.repository.id}\n')
      if args.non_dry_run:
        oss_fuzz_github.update_testcase_properties(testcase, issue.repository,
        testcase.put()

    monorail_issue = issue_tracker.get_original_issue(testcase.bug_information)
    if not monorail_issue.is_open:
       ...
```

Also, can we check if the GitHub issue is already closed here, to avoid `oss_fuzz_github.close_issue` retrieving the issue unnecessarily if we already know we don't need to do antyhing?